x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,217 @@
+    const newParams = { ...oldColumn.params };
+
+    if ('format' in newParams && field.type !== 'number') {
+      delete newParams.format;
+    }
+    return {
+      ...oldColumn,
+      dataType: field.type as DataType,
+      label: ofName(field.displayName),
+      sourceField: field.name,
+      params: newParams,
+    };
+  },
+  getPossibleOperationForField: ({ aggregationRestrictions, type }) => {
+    if (supportedTypes.has(type) && !aggregationRestrictions) {
+      return { dataType: type as DataType, isBucketed: false, scale: 'ordinal' };
+    }
+  },
+  getDisabledStatus(indexPattern: IndexPattern) {
+    const hasDateFields = indexPattern && getDateFields(indexPattern).length;
+    if (!hasDateFields) {
+      return i18n.translate('xpack.lens.indexPattern.lastValue.disabled', {
+        defaultMessage: 'This function requires the presence of a date field in your index',
+      });
+    }
+  },
+  buildColumn({ field, previousColumn, indexPattern }) {
+    const sortField =
+      indexPattern.timeFieldName ||
+      indexPattern.fields.find((f) => f.type === 'date')?.name ||
+      field.name;
+
+    const sortOrder =
+      (previousColumn?.params &&
+        'sortOrder' in previousColumn.params &&
+        previousColumn?.params?.sortOrder) ||
+      'desc';
+
+    return {
+      label: ofName(field.displayName),
+      dataType: field.type as DataType,
+      operationType: 'last_value',
+      isBucketed: false,
+      scale: 'ratio',
+      sourceField: field.name,
+      params: {
+        sortOrder,
+        sortField,
+      },
+    };
+  },
+  toEsAggsConfig: (column, columnId) => ({
+    id: columnId,
+    enabled: true,
+    schema: 'metric',
+    type: 'top_hits',
+    params: {
+      field: column.sourceField,
+      aggregate: 'concat',
+      size: 1,
+      sortOrder: column.params.sortOrder,
+      sortField: column.params.sortField,
+    },
+  }),
+
+  isTransferable: (column, newIndexPattern) => {
+    const newField = newIndexPattern.getFieldByName(column.sourceField);
+    return Boolean(newField && newField.type === column.dataType);
+  },
+
+  paramEditor: ({ state, setState, currentColumn, layerId }) => {
+    const dateFields = getDateFields(state.indexPatterns[state.layers[layerId].indexPatternId]);
+    const sortOrderButtons = [
+      {
+        id: `lns-lastValue-ascending`,
+        label: 'Ascending',
+        value: 'asc',
+      },
+      {
+        id: `lns-lastValue-descending`,
+        label: 'Descending',
+        value: 'desc',
+      },
+    ];
+    return (
+      <>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.lastValue.sortField', {
+            defaultMessage: 'Sort by date field',
+          })}
+          display="columnCompressed"
+          fullWidth
+        >
+          <EuiSelect
+            compressed
+            data-test-subj="lns-indexPattern-lastValue-sortField"
+            options={dateFields?.map((field: IndexPatternField) => {
```

Comment:
This date list doesn't differentiate empty and available date fields which might be helpful in some cases. Not necessarily required for the first iteration, but did we discuss this?

Reply:
Hmm, I don't recall discussing it, but good point. Let's talk during the sync what we want to do with it.

Reply:
We've decided that for the first generation we're leaving it as it is.